components/browser/engine-gecko/src/test/java/mozilla/components/browser/engine/gecko/prompt/GeckoPromptDelegateTest.kt

```diff
@@ -1436,6 +1438,96 @@ class GeckoPromptDelegateTest {
         verify(geckoResult, never()).complete(any())
     }
 
+    @Test
+    fun `WHEN onAddressSelect is called THEN SelectAddress prompt request must be provided with the correct callbacks`() {
+        val mockSession = GeckoEngineSession(runtime)
+
+        var isOnConfirmCalled = false
+        var isOnDismissCalled = false
+
+        var selectAddressPrompt: PromptRequest.SelectAddress = mock()
+
+        val promptDelegate = spy(GeckoPromptDelegate(mockSession))
+
+        // Capture the SelectAddress prompt request
+        mockSession.register(object : EngineSession.Observer {
+            override fun onPromptRequest(promptRequest: PromptRequest) {
+                selectAddressPrompt = promptRequest as PromptRequest.SelectAddress
+            }
+        })
+
+        val address = Address(
+            guid = "1",
+            givenName = "Firefox",
+            additionalName = "-",
+            familyName = "-",
+            organization = "-",
+            streetAddress = "street",
+            addressLevel3 = "address3",
+            addressLevel2 = "address2",
+            addressLevel1 = "address1",
+            postalCode = "1",
+            country = "Country",
+            tel = "1",
+            email = "@"
+        )
+        val addressSelectOption =
+            Autocomplete.AddressSelectOption(address.toAutocompleteAddress())
+
+        var geckoPrompt =
+            geckoSelectAddressPrompt(arrayOf(addressSelectOption))
+
+        var geckoResult = promptDelegate.onAddressSelect(
+            mock(),
+            geckoPrompt
+        )
+
+        // Verify that the onDismiss callback was called
+        geckoResult!!.accept {
+            isOnDismissCalled = true
+        }
+
+        selectAddressPrompt.onDismiss()
+        shadowOf(getMainLooper()).idle()
+        assertTrue(isOnDismissCalled)
+
+        // Verify that the onConfirm callback was called
+        geckoPrompt =
+            geckoSelectAddressPrompt(arrayOf(addressSelectOption))
+
+        geckoResult = promptDelegate.onAddressSelect(
+            mock(),
+            geckoPrompt
+        )
+
+        geckoResult!!.accept {
+            isOnConfirmCalled = true
+        }
+
+        selectAddressPrompt.onConfirm(selectAddressPrompt.addresses.first())
+        shadowOf(getMainLooper()).idle()
+        assertTrue(isOnConfirmCalled)
+
+        // Verify that when the request is already completed, the onConfirm callback is called
+        //  but the request is not confirmed
```

Comment:
```suggestion
        // Verify that when the prompt request is already completed and onConfirm callback is called,
        // then onConfirm callback is not executed
```
I think this one needs to be rephrased since it isn't quite making sense to me